backend/inspirehep/editor/authorlist_utils.py

```diff
@@ -339,9 +392,26 @@ def authorlist(text):
     result = create_authors(text)
 
     if "authors" in result:
-        for fullname, author_affs in result["authors"]:
+        for values in result["authors"]:
+
+            try:
+                if len(values[2][0]) == 1:
+                    emails = values[2]
+                    author_ids = []
+                else:
+                    author_ids = values[2]
+                    emails = []
+            except IndexError:
+                author_ids = []
+            try:
+                emails = values[3]
+            except IndexError:
+                emails = []
```

Comment:
maybe instead of adding a tuple for author add a dict with keys `emails`, `ids`, `affiliations` etc. and then you can get values by keys instead of this indexing logic

Reply:
I would do that but the other tests require the ```parse_author()``` to return them in a list format with a tuple.

Reply:
Could I maybe change the tests to check a ```dict``` rather than a ```list``` when they test ```parse_authors()```?
The function is only called in the ```authorlist_utils.py``` by ```create_authors()``` and nothing else uses it.
What do you think @MJedr and @drjova ?

Reply:
Example return value from ```create_authors()``` that calls the ```parse_authors()```
To make it easier to handle the values in ```authorlist()```
```
expected = [
        {
            "fullname": "O. Buchmueller", 
            "affiliations": [
                "High Energy Physics Group, Blackett Laboratory, Imperial College, Prince Consort Road, London SW7 2AZ, UK"
            ], 
            "ids": [], 
            "emails": []
        },
        {
            "fullname":  "K. Agashe", 
            "affiliations": [
                "University of Maryland, Department of Physics, College Park, MD 20742-4111, USA"
            ], 
            "ids": [], 
            "emails": []
        },
   ]
```

Reply:
Yes, then I could create the authors like this without any extra checks to determine what is what. That's why I would like to change the return value of ```parse_authors(text, affiliations)``` but then I will have to change all the tests for this function.

```
if "authors" in result:
        for author in result["authors"]:
            builder.add_author(
                builder.make_author(
                    author["fullname"], raw_affiliations=author["affiliations"], ids=author["ids"], emails=author["emails"]
                )
            )
        result["authors"] = builder.record["authors"]
```